src/controllers/customer.js

```diff
@@ -1,53 +1,118 @@
-        })
+        contact: {
+          create: {
+            phone,
+            email,
+          },
+        },
+      },
+      // We add an `include` outside of the `data` object to make sure the new contact is returned in the result
+      // This is like doing RETURNING in SQL
+      include: {
+        contact: true,
+      },
+    });
+
+    res.status(201).json({ customer: createdCustomer });
+  } catch (e) {
+    if (e instanceof Prisma.PrismaClientKnownRequestError) {
+      if (e.code === "P2002") {
+        return res
+          .status(409)
+          .json({ error: "A customer with the provided email already exists" });
+      }
     }
 
-    try {
-        /**
-         * This will create a Customer AND create a new Contact, then automatically relate them with each other
-         * @tutorial https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-related-record
-         */
-        const createdCustomer = await prisma.customer.create({
-            data: {
-                name,
-                contact: {
-                    create: {
-                        phone,
-                        email
-                    }
-                }
-            },
-            // We add an `include` outside of the `data` object to make sure the new contact is returned in the result
-            // This is like doing RETURNING in SQL
-            include: { 
-                contact: true
-            }
-        })
-
-        res.status(201).json({ customer: createdCustomer })
-    } catch (e) {
-        if (e instanceof Prisma.PrismaClientKnownRequestError) {
-            if (e.code === "P2002") {
-                return res.status(409).json({ error: "A customer with the provided email already exists" })
-            }
-        }
-        
-        res.status(500).json({ error: e.message })
+    res.status(500).json({ error: e.message });
+  }
+};
+
+const updateCustomer = async (req, res) => {
+  let { id } = req.params;
+  id = Number(id);
+
+  const findCustomer = await prisma.customer.findUnique({
+    where: {
+      id: id,
+    },
+    include: {
+      contact: true,
+    },
+  });
+
+  if (findCustomer) {
+    const { name, contact } = req.body;
+
+    let phone = undefined
+    let email = undefined
+
+    if (contact) {
+        phone = contact.phone;
+        email = contact.email;
+      }
+
+    if (!name && !phone && !email) {
+      return res.status(400).json({
+        error: "Missing fields in request body",
+      });
     }
-}
+    const updateCustomer = await prisma.customer.update({
+      where: {
+        id: id,
+      },
+      data: {
+        name: name,
+        contact: {
+          update: {
+            email: email,
+            phone: phone,
+          },
+        },
+      },
+      include: {
+        contact: true,
+        tickets: true,
+      },
+    });
+
+    res.status(201).json({ customer: updateCustomer });
+  } else {
+    return res.status(404).json({
```

Comment:
Now 404 will be returned for any other error  right ? is not really specified if customer does not exist. I may be understanding it wrong

Reply:
Only when customer does not exist. I have an if for a FindCutsomer